### PR TITLE
docs: fix examples for several rules

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fixed some examples I noticed while reviewing recent changes in the rules docs.

### CODE REVIEW

File: docs/src/rules/function-paren-newline.md [hunk 1]
```diff
@@ -339,11 +339,15 @@ var barbaz = function(
...
+  bar,
+  baz
+) => {};
 
-foo(bar,
-  baz);
+foo(
+  bar,
+  baz
+);
```
Comment (mdjermanovic) [inline docs/src/rules/function-paren-newline.md:350]: These examples in "incorrect" were "correct".

File: docs/src/rules/id-match.md [hunk 2]
```diff
@@ -143,7 +143,7 @@ Examples of **correct** code for this rule with the `"^[a-z]+([A-Z][a-z]+)*$", {
 ```js
 /*eslint id-match: [2, "^[a-z]+([A-Z][a-z]+)*$", { "onlyDeclarations": true }]*/
 
-do_something(__dirname);
+foo = __dirname;
```
Comment (mdjermanovic) [inline docs/src/rules/id-match.md:146]: This was "correct" regardless of the option, so I replaced it with an example that would be "incorrect" with `"onlyDeclarations": false`

File: docs/src/rules/indent.md [hunk 3]
```diff
@@ -192,7 +192,7 @@ Examples of **correct** code for this rule with the `4, { "ignoredNodes": ["Call
 foo();
 bar();
 
-})
+})();
```
Comment (mdjermanovic) [inline docs/src/rules/indent.md:195]: The selector targets IIFEs. This wasn't an IIFE, so it was "incorrect".

File: docs/src/rules/max-statements-per-line.md [hunk 4]
```diff
@@ -76,7 +76,7 @@ if (condition) { bar = 1; } else { baz = 2; }
 for (var i = 0; i < length; ++i) { bar = 1; baz = 2; }
 switch (discriminant) { case 'test': break; default: break; }
 function foo() { bar = 1; baz = 2; }
-var qux = function qux() { bar = 1; };
+var qux = function qux() { bar = 1; baz = 2; };
```
Comment (mdjermanovic) [inline docs/src/rules/max-statements-per-line.md:79]: Function expression doesn't count as a statement, so this was "correct".

File: docs/src/rules/max-statements.md [hunk 5]
```diff
@@ -99,12 +99,13 @@ function foo() {
   var foo7 = 7;
   var foo8 = 8;
   var foo9 = 9;
-  var foo10 = 10;
-  return function () {
+  return function () { // 10
```
Comment (mdjermanovic) [inline docs/src/rules/max-statements.md:102]: `return` itself was the 11th statement, so this was "incorrect".

File: docs/src/rules/one-var.md [hunk 6]
```diff
@@ -416,12 +416,16 @@ Examples of **correct** code for this rule with the `{ var: "never" }` option:
 /*eslint-env es6*/
 
 function foo() {
-    var bar,
-        baz;
-    const foobar = 1; // `const` and `let` declarations are ignored if they are not specified
+    var bar;
+    var baz;
```
Comment (mdjermanovic) [inline docs/src/rules/one-var.md:420]: This was "incorrect".

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**